### ISSUE TITLE
Format error in python 2.6

### ISSUE THREAD

**Issue by vwbusguy** [cameronbwhite/Flask-CAS#16]
Here's the trace on login route.  It actually seems to happen anytime .format() is used.  For some reason it works in Python 2.7 but not the Python 2.6.6 shipped in CentOS6.  The issue is that Python 2.6 expects a key in the {} for format (https://docs.python.org/2/whatsnew/2.6.html#pep-3101-advanced-string-formatting).  

Here's the trace:
```
127.0.0.1 - - [15/Oct/2014 17:17:40] "GET /auth-cas/login/ HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask_cas/routing.py", line 35, in login
    flask.url_for('.login', _external=True))
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask_cas/cas_urls.py", line 72, in create_cas_login_url
    ('renew',renew),
  File "/home/sawilliams/auth-cas/env/lib/python2.6/site-packages/flask_cas/cas_urls.py", line 46, in create_url
    url = urljoin(url, '?{}'.format(urlencode(list(query))))
ValueError: zero length field name in format
```

I should have a pull request that will fix this soon.  Basically, since we're only using format to cast one string, using %s vs .format() is trivial, but will allow for better compatibility as it's still valid in py3.  Replacing this with a %s cast makes it work on EL6.

Scott

**Comment by cameronbwhite** [comment]
Sorry I have been ignoring this project for a while. It looks like this problem is no longer reproducible  on `master` or `cas-version-2-and-3`. 

:smile_cat: Thanks for the nice words :+1:

**Comment by vwbusguy** [comment]
Hey there,

Just wanted to report that this is indeed fixed in our EL6 (Python 2.6) environments.  Many thanks!

Scott
